fdbclient/include/fdbclient/TenantEntryCache.actor.h

```diff
@@ -311,15 +493,22 @@ class TenantEntryCache : public ReferenceCounted<TenantEntryCache<T>>, NonCopyab
 
 		// Launch reaper task to periodically refresh cache by scanning database KeyRange
 		TenantEntryCacheRefreshReason reason = TenantEntryCacheRefreshReason::PERIODIC_TASK;
+		Future<Void> initalWatchFuture = Void();
+		lastTenantIdRefresher = lastTenantIdWatch(this);
 		if (refreshMode == TenantEntryCacheRefreshMode::PERIODIC_TASK) {
 			refresher = recurringAsync([&, reason]() { return refresh(reason); },
 			                           CLIENT_KNOBS->TENANT_ENTRY_CACHE_LIST_REFRESH_INTERVAL, /* interval */
 			                           true, /* absoluteIntervalDelay */
 			                           CLIENT_KNOBS->TENANT_ENTRY_CACHE_LIST_REFRESH_INTERVAL, /* intialDelay */
 			                           TaskPriority::Worker);
+		} else if (refreshMode == TenantEntryCacheRefreshMode::WATCH) {
+			initalWatchFuture = setInitalWatch.getFuture();
```

Comment:
Not sure if this is right approach, based on this code, the caller would end up waiting till the first `tenant` creation? (Given the `setInitialWatch` promise is released when watch is triggered..); if yes, would that be a desired behavior..

Am not sure if we need to track `intialWatch` trigger is needed, we need not to change the `init` semantics, as part of startup, we can read the tenant-subspace keyrange and caller if wishes can wait for range-read completion, however, watch is set such that any future updates will be tracked accordingly. 
Thoughts?

Reply:
Actually `setInitialWatch` is released right after the watch is set, not triggered. This will ensure that the actor at least sets the watch during init. Otherwise in testing i found that sometimes we time out because we don't even set the watch (there are other tasks that run first) and the watch ends up being created much later. I think this is a small wait for the client so it shouldn't add much overhead

Reply:
> This will ensure that the actor at least sets the watch during init. Otherwise in testing i found that sometimes we time out because

Initially the caller was waiting for database read-range to finish and it seems highly unlikely that `watch would` not be set even after database read is done? Does the test needs update instead of making the code to work better for the test?

Reply:
Oh sorry i forgot, the use case for this is mainly simulation. Currently i want to test that when a tenant is created the watch is triggered. However i need to make sure the watch is set up before the tenant is created, hence i use the promise. I don't know if you know of a cleaner way to do this but i couldn't really think of any (maybe the other one is waiting on some bool to change in the test). I can guard this so it only happens in simulation, thoughts?

Reply:
Did you try updating the sequence such as: launch actor to set the `watch` followed by the actor launched to do the `initial database load` 

In the test (as done in other places too) if the test waits on `init()` future, then it would guanratee the watch being set?

Reply:
Discussed offline, will add option to init to wait for the initial watch to be set